src/main/java/com/ase/restservice/service/FinanceService.java

```diff
@@ -18,15 +23,61 @@ public class FinanceService implements FinanceServiceI {
      */
     public boolean isStockIdValid(String stockId) {
         try {
-            Stock apiStock = YahooFinance.get(stockId);
+            yahoofinance.Stock apiStock = YahooFinance.get(stockId);
             if (apiStock == null) {
                 return false;
             }
             return true;
         } catch (IOException e) {
             e.printStackTrace();
         }
-
         return false;
     }
+
+    /**
+     * Gets the price for a given stock ID.
+     *
+     * @param stockId Stock ID
+     * @return Float real-time value of the stock
+     * @throws ResourceNotFoundException if the stock ID is invalid
+     * @throws IOException when there is a connection error
+     */
+    public Float getStockPrice(String stockId) throws ResourceNotFoundException, IOException {
+        // Check if the stock ID is valid
+        if (!isStockIdValid(stockId)) {
+            throw new ResourceNotFoundException(
```

Comment:
Can we create a new custom error for 400 BadRequestException and throw that here since the failure of isStockIdValid() means there is an error in the client's request? (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400)

Reply:
Yes, will address in the next iteration (I can pick up the issue `[T5] Create more custom exceptions`)

Reply:
In fact, maybe we can just throw that error in isStockIdValid() method. So, any method calling it that would also have that behavior